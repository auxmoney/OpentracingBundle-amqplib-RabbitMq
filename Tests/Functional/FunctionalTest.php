<?php

declare(strict_types=1);

namespace Auxmoney\OpentracingAmqplibRabbitMqBundle\Tests\Functional;

use Auxmoney\OpentracingBundle\Tests\Functional\JaegerConsoleFunctionalTest;
use Symfony\Component\Process\Process;
use Symfony\Component\Yaml\Yaml;

class FunctionalTest extends JaegerConsoleFunctionalTest
{
    public function testTraceIncludesMessageConsumer(): void
    {
        $this->setUpTestProject('default');

        $this->runInTestProject(['symfony', 'console', 'rabbitmq:setup-fabric']);

        $process = new Process(['symfony', 'console', 'test:message'], 'build/testproject');
        $process->mustRun();
        $output = $process->getOutput();
        $traceId = substr($output, 0, strpos($output, ':'));
        self::assertNotEmpty($traceId);

        $this->runInTestProject(['symfony', 'console', 'rabbitmq:consumer', '-m', '1', 'test_queue']);

        $spans = $this->getSpansFromTrace($this->getTraceFromJaegerAPI($traceId));
        self::assertCount(4, $spans);

        $traceAsYAML = $this->getSpansAsYAML($spans, '[].{operationName: operationName, startTime: startTime, spanID: spanID, references: references, tags: tags[?key==\'command.exit-code\' || key==\'test.tag\'].{key: key, value: value}}');
        self::assertStringEqualsFile(__DIR__ . '/FunctionalTest.expected.yaml', $traceAsYAML);
    }
}
