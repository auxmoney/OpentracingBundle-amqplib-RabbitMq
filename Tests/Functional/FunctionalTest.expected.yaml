operationName: 'test:message'
tags:
  -
    key: test.tag
    value: command
  -
    key: command.exit-code
    value: 0
children:
  -
    operationName: 'RabbitMq: Publishing message to "test_queue" exchange'
    tags: {  }
    children:
      -
        operationName: 'RabbitMq: Processing message from "test_queue" queue'
        tags: {  }
        children:
          -
            operationName: 'manual consumer span'
            tags:
              -
                key: test.tag
                value: manual
